civicrm.module

```diff
@@ -172,15 +172,19 @@ function civicrm_menu_alter(&$items) {
 /**
  * Implements hook_layout_access_info().
  *
- * Adding a CiviCRM checksum visibility condition.
+ * Adding a CiviCRM checksum visibility condition and a URL condition that
+ * considers the query parameters in the URL.
  */
 function civicrm_layout_access_info() {
-  $info = array(
-    'civicrm_checksum' => array(
-      'title' => t('CiviCRM checksum'),
-      'description' => t('Has the page been loaded with a valid CiviCRM checksum?'),
-      'class' => 'CivicrmChecksumLayoutAccess',
-    ),
+  $info['civicrm_checksum'] = array(
+    'title' => t('CiviCRM checksum'),
+    'description' => t('Has the page been loaded with a valid CiviCRM checksum?'),
+    'class' => 'CivicrmChecksumLayoutAccess',
+  );
+  $info['civicrm_urls'] = array(
+    'title' => t('CiviCRM URLs'),
+    'description' => t('Control access by the path and query parameters of the current CiviCRM page.'),
```

Comment:
I think this might need a bit more explanation to differentiate it from "CiviCRM pages". Though we should probably also improve the description there as well.

Reply:
Seems like the current description "Control access by the path and query parameters of the current CiviCRM page" is pretty concise. But I would welcome suggestions for replacement text. (Where does this description actually show up?)

By contrast, the description for "CiviCRM pages" is "Control access by the current CiviCRM page," which is a bit opaque. How about changing this to "Control access to the current CiviCRM page by whether it is admin or public"?

I might also suggest that changes to the "CiviCRM pages" visibility condition should be a separate issue and PR (?).

Reply:
Just looking at this again. Let's keep this description as is. Though I think we could make the title more like Backdrop core language, which uses "URL path". Let's change the title to "CiviCRM URL path and query". Note that it should be singular.

Reply:
Done.